cocos/2d/CCNode.h

```diff
@@ -950,8 +955,8 @@ class CC_DLL Node : public Ref
             return (n1->_localZOrderAndArrival < n2->_localZOrderAndArrival);
         });
 #else
-        std::stable_sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
-            return n1->_localZOrder < n2->_localZOrder;
+        std::sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
+            return n1->_localZOrder < n2->_localZOrder || (n1->_localZOrder == n2->_localZOrder && n1->_getOrderOfArrival() < n2->_getOrderOfArrival() );
```

Comment:
The codes looks good, but i am not sure the performance. May be we should keep `_orderArrival` on 32bit.

Reply:
Yes, direct access the variable maybe more better.

Reply:
Yep, the bug is imported to improve performance for 64bit devices, but we can not slow down 32bit devices.

Reply:
Let's change something.